Feat/user crud

EasySouls opened the pull request:
I added the basic crud functionalities and endpoints for the UserModel, aside from the authentication.
I thought we could separate that to another module, but we'll see.
I also created a PrismaService, which lets us use the PrismaClient through dependency injection.

Fixed a module import with PrismaService, by making the methods in the UserService class async.
And also set up the OpenApi basics, although without precise information yet.
Resolves #9

EasySouls committed:
Add schemas for ratings

EasySouls committed:
User crud setup

EasySouls committed:
Implement basic crud methods for user and add openapi config

EasySouls committed:
Fix module import error with PrismaService

FearsomeRover commented:
The swagger config is great, but we have only one problem with it. If we are using prisma generated classes, we wont be able to use those types as examples provided in the swagger config. So as painful as it looks, we need to write our own dto-s. They can be identical to the prisma generated ones. If you need any help doing this, just let me know. You can see examples in [startlap](https://github.com/kir-dev/startlap-sch). The usual way of going about this, is to create an entiy, and then using omittype and partialtype create classes for creating and updating the user. 
I added introspectComments to the nest cli, so we can also create example values for the requests. [example](https://api.startlap.kir-dev.hu/api#/default/CollectionController_create) like it is seen for example here.
The great thing about creating our own classes, is that we can also use class-validator, to check for example if the given email is valid.
But great work so far, keep it up.

FearsomeRover commented on apps/backend/src/users/users.controller.ts:
Here is where we need to import our own classes instead of the prisma generated ones

FearsomeRover committed:
added introspect comments

EasySouls committed:
Created User entity and dto objects

EasySouls committed:
Prettier fix

EasySouls commented:
@FearsomeRover I think I implemented everything you asked for. The dtos didn't show on the openapi page, but I added @ApiProperty decorators and now they work.
Also added a @default(uuid()) to the authSchId parameter in the schema, so that it would be compatible with the entity. I don't know yet how authsch works, but im sure we'll get that later via a session, cookie or something like that, when we have auth implemented.

FearsomeRover reviewed (approved):
Nice job! We can keep the default uuid for now, but later we will get thet id from authsch. We can do the authentication together in a workshop, as it is quite complex i think.
